data/mods/Magiclysm/eoc_spell_learn_boost.json

```diff
@@ -0,0 +1,185 @@
+[
+  {
+    "type": "jmath_function",
+    "id": "mc_spell_exp_diff",
+    "num_args": 1,
+    "//": "accept the spell level, return a difference in experience between spell's current level and the next level",
+    "return": "spell_exp(_0 + 1) - spell_exp(_0)"
+  },
+  {
+    "type": "jmath_function",
+    "id": "mc_spell_train_focus_factor",
+    "num_args": 2,
+    "//args": "_0=focus, _1=spell_level",
+    "//": "At level 15 focus factor is enabled, 50 focus = twice more spellcasts, 100 focus = same, 200 focus = twice less spellcasts",
+    "return": "_1 > 15 ? (max(20, _0) - 50) / 100 + 0.5 : 1"
+  },
+  {
+    "type": "jmath_function",
+    "id": "mc_spell_cost_train_factor",
+    "num_args": 3,
+    "//args": "_0=cost,_1=min_spellcasts,_2=max_spellcasts",
+    "//": "accept the spell _cost, return how many spellcasts are required to level the spell; formula assumes 2 spellcasts at 100% manapool cost, and 10 spellcasts at 1% manapool cost",
+    "return": "max(_1, min(_2, (_2 - (_1 * max(0, _0) / u_val('mana_max') * _2 / _1 ))))"
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_MC_CASTED_A_SPELL_LEARN_BOOST",
+    "eoc_type": "EVENT",
+    "required_event": "character_casts_spell",
+    "//": "Regular classes levels some faster and needs 2-10 spellcasts to level up depends on spell cost",
+    "condition": {
+      "or": [
+        { "compare_string": [ "MANA_CRYST_MANA", { "context_val": "school" } ] },
+        { "compare_string": [ "MANA_SEEKER_BOLTS", { "context_val": "school" } ] },
+        { "compare_string": [ "ANIMIST", { "context_val": "school" } ] },
+        { "compare_string": [ "BIOMANCER", { "context_val": "school" } ] },
+        { "compare_string": [ "DRUID", { "context_val": "school" } ] },
+        { "compare_string": [ "EARTHSHAPER", { "context_val": "school" } ] },
+        { "compare_string": [ "KELVINIST", { "context_val": "school" } ] },
+        { "compare_string": [ "MAGUS", { "context_val": "school" } ] },
+        { "compare_string": [ "STORMSHAPER", { "context_val": "school" } ] },
+        { "compare_string": [ "TECHNOMANCER", { "context_val": "school" } ] },
+        { "compare_string": [ "ALCHEMIST", { "context_val": "school" } ] }
+      ]
+    },
+    "effect": [
+      {
+        "math": [
+          "u_spell_exp(_spell)",
+          "+=",
+          "mc_spell_exp_diff(u_spell_level(_spell)) / mc_spell_cost_train_factor(_cost, 2, 10) * mc_spell_train_focus_factor(u_val('focus'),u_spell_level(_spell))"
+        ]
+      }
+    ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_MC_CASTED_A_SPELL_LEARN_BOOST_ATTUNED",
+    "eoc_type": "EVENT",
+    "required_event": "character_casts_spell",
+    "//": "Attuned classes levels faster since they have no any books, and needs 1-5 spellcasts to level up depends on spell cost",
+    "condition": {
+      "or": [
+        {
+          "compare_string": [ "NONE", { "context_val": "school" } ],
```

Comment:
I'm still learning debug spells even with this version of the file, and doing a little testing revealed that this line is the culprit. Also since this EOC is for attuned classes only, you should probably remove the regular classes from the list, they're currently getting tripled exp from the other EOC and this one combined.

Reply:
Opps №2... Fixed.